src/stories/SelectionList.stories.tsx

```diff
@@ -101,42 +105,38 @@ Default.args = {
     initiallyFocusedOptionKey: 'option-2',
 };
 
-function WithTextInput(args) {
+function WithTextInput(props: BaseSelectionListProps<ListItem>) {
     const [searchText, setSearchText] = useState('');
     const [selectedIndex, setSelectedIndex] = useState(1);
 
-    const sections = _.map(args.sections, (section) => {
-        const data = _.reduce(
-            section.data,
-            (memo, item, index) => {
-                if (!item.text.toLowerCase().includes(searchText.trim().toLowerCase())) {
-                    return memo;
-                }
-
-                const isSelected = selectedIndex === index + section.indexOffset;
-                memo.push({...item, isSelected});
+    const sections = props.sections.map((section) => {
+        const data = section.data.reduce<Array<ListItem & {isSelected: boolean}>>((memo, item, index) => {
```

Comment:
Let's create a type for it `ListItem & {isSelected: boolean}`. Also you can check, maybe you can reuse existing one

Reply:
Actually, it turns out that just `ListItem` is enough!